framework/decode/vulkan_offscreen_swapchain.cpp

```diff
@@ -103,38 +103,49 @@ VkResult VulkanOffscreenSwapchain::CreateSwapchainKHR(VkResult
     if (insert_frame_boundary_)
     {
         VkResult result;
+        command_pools_.resize(device_info->queue_family_index_enabled.size());
+        command_buffers_.resize(device_info->queue_family_index_enabled.size());
 
-        VkCommandPoolCreateInfo commandPoolCreateInfo;
-        commandPoolCreateInfo.sType            = VK_STRUCTURE_TYPE_COMMAND_POOL_CREATE_INFO;
-        commandPoolCreateInfo.pNext            = nullptr;
-        commandPoolCreateInfo.queueFamilyIndex = default_queue_family_index_;
-        commandPoolCreateInfo.flags            = 0;
-
-        result = device_table_->CreateCommandPool(device, &commandPoolCreateInfo, nullptr, &command_pool_);
-        GFXRECON_ASSERT(result == VK_SUCCESS);
-
-        VkCommandBufferAllocateInfo commandBufferAllocateInfo;
-        commandBufferAllocateInfo.sType              = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
-        commandBufferAllocateInfo.pNext              = nullptr;
-        commandBufferAllocateInfo.commandPool        = command_pool_;
-        commandBufferAllocateInfo.level              = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
-        commandBufferAllocateInfo.commandBufferCount = 1;
-
-        result = device_table_->AllocateCommandBuffers(device, &commandBufferAllocateInfo, &command_buffer_);
-        GFXRECON_ASSERT(result == VK_SUCCESS);
-
-        VkCommandBufferBeginInfo commandBufferBeginInfo;
-        commandBufferBeginInfo.sType            = VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO;
-        commandBufferBeginInfo.pNext            = nullptr;
-        commandBufferBeginInfo.flags            = 0;
-        commandBufferBeginInfo.pInheritanceInfo = nullptr;
-
-        result = device_table_->BeginCommandBuffer(command_buffer_, &commandBufferBeginInfo);
-        GFXRECON_ASSERT(result == VK_SUCCESS);
-
-        result = device_table_->EndCommandBuffer(command_buffer_);
-        GFXRECON_ASSERT(result == VK_SUCCESS);
-
+        for (uint32_t family_index = 0; family_index < device_info->queue_family_index_enabled.size(); family_index++)
```

Comment:
Just add a comment above this explaining why your doing this so people can understand what's going on.  Just something similar to the first sentence of your commit is fine.

Reply:
Done =)